src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.serviceimpl.FruitTransactionValidatorImpl;
+import java.util.Optional;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private FruitTransactionValidatorImpl validator;
+
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        validator.validateOperation(fruitTransaction);
+        String fruit = fruitTransaction.getFruit();
+        validator.validateFruit(fruit);
+        int amountToPurchase = fruitTransaction.getQuantity();
+        validator.validateAmount(amountToPurchase, FruitTransaction.Operation.PURCHASE);
+
+        Optional<Integer> currentBalance = Optional.ofNullable(Storage.fruitStorage.get(fruit));
+        int balanceAfterPurchase = currentBalance.orElse(0) - amountToPurchase;
+
+        if (balanceAfterPurchase < 0) {
+            throw new RuntimeException("Insufficient stock for item " + fruit);
+        }
+        Storage.fruitStorage.put(fruit, balanceAfterPurchase);
+    }
```

Comment:
The `apply` method is doing validation and also updating the storage. This violates the Single Responsibility Principle. Consider refactoring the code to separate concerns, for example, by moving validation logic to a separate method or class.